### PR TITLE
display-test-app iOS build updates

### PR DESCRIPTION
**Pull request by toddsouthenbentley**
Added build:ios-app script and fixed build warnings.

### CODE REVIEW

**Comment by pmconne** [comment]
@markschlosseratbentley please review.

**Comment by markschlosseratbentley** [comment]
> @markschlosseratbentley please review.

@pmconne @toddsouthenbentley Tested it -- works for me.

**Review by markschlosseratbentley — APPROVED** [review]

File: test-apps/display-test-app/package.json
Comment (markschlosseratbentley) [inline test-apps/display-test-app/package.json:36]: Should we add 'npm run build:ios-app' to the new dta iOS CI job step as part of this PR?

Comment (toddsouthenbentley) [reply test-apps/display-test-app/package.json:36]: I don't know. The Xcode project builds quite quickly once it has downloaded the add-on binary but takes a long time to do that. If each CI job would have to download the add-on then I'd say we probably don't want to do this.

Comment (markschlosseratbentley) [reply test-apps/display-test-app/package.json:36]: @calebmshafer @aruniverse do you have any thoughts on this?

**Review by tcobbs-bentley — APPROVED** [review]

File: test-apps/display-test-app/package.json
Comment (aruniverse) [reply test-apps/display-test-app/package.json:36]: If it takes a while to download the add-on and what not, then id rather not have this be apart of the ci, maybe once a pr is merged into master? How often will the swift code change? How often does the swift side break?

Comment (markschlosseratbentley) [reply test-apps/display-test-app/package.json:36]: I've definitely had the swift side break.  Not sure how often, but it's happened.

Comment (toddsouthenbentley) [reply test-apps/display-test-app/package.json:36]: The swift side should only break if changes were made to add-on's iOS API.

Comment (markschlosseratbentley) [reply test-apps/display-test-app/package.json:36]: Yeah, I think that's why it has broken in the past.  So that's fair, that we don't need to do it every PR.
Were you planning on making the app open up and successfully download an iModel as part of the CI?  If so, wouldn't we need to build the swift side anyway?

Comment (aruniverse) [reply test-apps/display-test-app/package.json:36]: Hopefully when someone changes the mobile addon, theyd test the full app and we'd avoid having to ensure it in the ci here, but ill defer to caleb.

Comment (toddsouthenbentley) [reply test-apps/display-test-app/package.json:36]: Yes, we'll need to fully build the app if we want to run it in the simulator in some sort of automated fashion. If we add that to CI then we'd definitely want to cache the add-ons.

Comment (markschlosseratbentley) [reply test-apps/display-test-app/package.json:36]: > Hopefully when someone changes the mobile addon, theyd test the full app and we'd avoid having to ensure it in the ci here, but ill defer to caleb.

In the past, it has just broken.  So we either need a release checklist which includes building + running display-test-app, or automate building + running it regularly somehow.

Comment (aruniverse) [reply test-apps/display-test-app/package.json:36]: Do we just add it to the [release pre-reqs](https://dev.azure.com/bentleycs/iModelTechnologies/_wiki/wikis/iModelTechnologies.wiki/8499/Process?anchor=initial-steps)?

Comment (markschlosseratbentley) [reply test-apps/display-test-app/package.json:36]: Yes, perhaps.  I suppose this should not hold up this PR --- especially @toddsouthenbentley if you are going to look at caching the addon for running DTA in the simulator for CI in a future PR

**Pull request merged**
